src/server/player_sao.cpp

```diff
@@ -519,12 +519,14 @@ void PlayerSAO::rightClick(ServerActiveObject *clicker)
 
 void PlayerSAO::setHP(s32 target_hp, const PlayerHPChangeReason &reason, bool from_client)
 {
+	s32 hp_change = std::max<s32>(target_hp, -U16_MAX) - (s32)m_hp;
+
 	target_hp = rangelim(target_hp, 0, U16_MAX);
```

Comment:
could fix that test case by replacing this with `std::max(target_hp, 0)`

Reply:
It will break again delivering more damage than hp to hpchange callback.

Reply:
the second one, not the first one